### PR TITLE
feat(editor): Execution filter

### PR DESCRIPTION
**Pull request by cstuncsik**
PAY-149

### CODE REVIEW

File: packages/cli/src/WorkflowExecuteAdditionalData.ts [hunk 1]
```diff
@@ -264,6 +264,19 @@ async function pruneExecutionData(this: WorkflowHooks): Promise<void> {
 	}
 }
 
+async function saveExecutionMetadata(
+	executionId: string,
+	executionMetadata: Record<string, string>,
+): Promise<void> {
+	for (const [key, value] of Object.entries(executionMetadata)) {
+		await Db.collections.ExecutionMetadata.save({
```
Comment (flipswitchingmonkey) [inline packages/cli/src/WorkflowExecuteAdditionalData.ts:272]: Can we do this as a transaction or a Promise.all()? Otherwise we'll introduce multiple delays here.

File: packages/cli/src/executions/executions.service.ts [hunk 2]
```diff
@@ -84,17 +105,40 @@ export class ExecutionsService {
...
+				for (const md of metadata) {
+					query = query.andWhere('md.key = :key AND md.value = :value', md);
+				}
+			}
+
+			if (filteredFields.length > 0) {
+				query = query.andWhere(countFilter);
+			}
+
+			// const countParams = { where: { workflowId: In(sharedWorkflowIds), ...countFilter } };
```
Comment (flipswitchingmonkey) [inline packages/cli/src/executions/executions.service.ts:142]: can we remove these comments?

**Review by flipswitchingmonkey — CHANGES_REQUESTED** [review]

File: packages/cli/src/databases/entities/ExecutionMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	@ManyToOne('ExecutionEntity', 'metadata', {
+		onDelete: 'CASCADE',
+	})
+	execution: ExecutionEntity;
+
+	@RelationId((executionMetadata: ExecutionMetadata) => executionMetadata.execution)
+	executionId: number;
+
+	@Column('text')
+	key: string;
```
Comment (netroy) [inline packages/cli/src/databases/entities/ExecutionMetadata.ts:18]: we should either use stricter typing on this column, and add an index for faster searching, or get move this data to a simple json column on the executions table. 
in this current form, the additional table/entity feels unnecessary.

Comment (netroy) [reply packages/cli/src/databases/entities/ExecutionMetadata.ts:18]: let me rephrase: Does this really need to be a separate table?

Comment (krynble) [reply packages/cli/src/databases/entities/ExecutionMetadata.ts:18]: Yes, I think so - you can save multiple key-value pairs for each execution. The goal is to allow users to save and read data during execution and afterwards, search for this data in the executions panel.

When querying, you can query by some of the keys (initial implementation is a single key-value search but this can be changed in the future)

Screenshot 2023-03-16 at 13 28 52

**Review by krynble — APPROVED** [review]

**Review by krynble — APPROVED** [review]

**Review by flipswitchingmonkey — APPROVED** [review]

**Comment by csuermann** [comment]
Let me repeat my proposal from a few weeks ago: Migrations affecting the potentially gigantic executions_entity table are generally very problematic because of the sheer volume of that table. Therefore, we should consider splitting that table in two: In one we only store the execution data blob, in the other all the other (lightweight) columns. JOINing them when needed is still straightforward and future migrations should no longer be problematic. Thoughts?

**Comment by krynble** [comment]
I like the idea @csuermann , this might greatly speed things up.
I'll try creating a PR about this and see how this goes.

**Comment by janober** [comment]
Got released with [`n8n@0.222.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.222.0)
